src/libraries/System.Private.CoreLib/src/System/Text/Ascii.Utility.cs

```diff
@@ -94,613 +71,275 @@ private static bool FirstCharInUInt32IsAscii(uint value)
-            pBuffer += CountNumberOfLeadingAsciiBytesFromUInt32WithSomeNonAsciiData(currentDWord);
-
-            goto Finish;
-
-        InputBufferLessThanOneVectorInLength:
-
-            // These code paths get hit if the original input length was less than one vector in size.
-            // We can't perform vectorized reads at this point, so we'll fall back to reading primitives
-            // directly. Note that all of these reads are unaligned.
-
-            Debug.Assert(bufferLength < SizeOfVector128);
-
-            // QWORD drain
-
-            if ((bufferLength & 8) != 0)
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            static bool Test512(Vector512<byte> value)
             {
-                if (UIntPtr.Size == sizeof(ulong))
-                {
-                    // If we can use 64-bit tzcnt to count the number of leading ASCII bytes, prefer it.
-
-                    ulong candidateUInt64 = Unsafe.ReadUnaligned<ulong>(pBuffer);
-                    if (!AllBytesInUInt64AreAscii(candidateUInt64))
-                    {
-                        // Clear everything but the high bit of each byte, then tzcnt.
-                        // Remember to divide by 8 at the end to convert bit count to byte count.
-
-                        candidateUInt64 &= UInt64HighBitsOnlyMask;
-                        pBuffer += (nuint)(BitOperations.TrailingZeroCount(candidateUInt64) >> 3);
-                        goto Finish;
-                    }
-                }
-                else
+                if (Avx512F.IsSupported)
                 {
-                    // If we can't use 64-bit tzcnt, no worries. We'll just do 2x 32-bit reads instead.
-
-                    currentDWord = Unsafe.ReadUnaligned<uint>(pBuffer);
-                    uint nextDWord = Unsafe.ReadUnaligned<uint>(pBuffer + 4);
-
-                    if (!AllBytesInUInt32AreAscii(currentDWord | nextDWord))
-                    {
-                        // At least one of the values wasn't all-ASCII.
-                        // We need to figure out which one it was and stick it in the currentMask local.
-
-                        if (AllBytesInUInt32AreAscii(currentDWord))
-                        {
-                            currentDWord = nextDWord; // this one is the culprit
-                            pBuffer += 4;
-                        }
-
-                        goto FoundNonAsciiDataInCurrentDWord;
-                    }
+                    return value.ExtractMostSignificantBits() != 0;
                 }
 
-                pBuffer += 8; // successfully consumed 8 ASCII bytes
+                return Test256(value.GetLower() | value.GetUpper());
             }
+        }
 
-            // DWORD drain
-
-            if ((bufferLength & 4) != 0)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static nuint ComputeIndex(Vector128<byte> value)
```

Comment:
These should use internal helpers on `VectorXXX` that match the naming approved in https://github.com/dotnet/runtime/issues/98055#issuecomment-2032720254

In this case it looks to be `IndexOfWhereAllBitsSet(value)`

Reply:
Sure, I'll rename them.